Objective-C/Tests/QueryTest+Main.m

```diff
@@ -1873,5 +1794,280 @@ - (void) validateN1QLQuery: (NSString*)queryString {
     AssertEqualObjects([result[1] stringForKey: @"lastName"], @"Ice Cream");
 }
 
+#pragma mark - LiveQuery
+
+- (void) testLiveQuery {
+    [self loadNumbers: 100];
+    
+    __block int count = 0;
+    XCTestExpectation* x = [self expectationWithDescription: @"changes"];
+    CBLQuery* q = [CBLQueryBuilder select: @[kDOCID]
+                                     from: [CBLQueryDataSource database: self.db]
+                                    where: [[CBLQueryExpression property: @"number1"] lessThan: [CBLQueryExpression integer: 10]]
+                                  orderBy: @[[CBLQueryOrdering property: @"number1"]]];
+    
+    id token = [q addChangeListener: ^(CBLQueryChange* change) {
+        count++;
+        AssertNotNil(change.query);
+        AssertNil(change.error);
+        NSArray<CBLQueryResult*>* rows = [change.results allObjects];
+        if (count == 1) {
+            AssertEqual(rows.count, 9u);
+        } else {
+            AssertEqual(rows.count, 10u);
+            CBLDocument* doc = [self.db documentWithID: [rows[0] valueAtIndex: 0]];
+            AssertEqualObjects([doc valueForKey: @"number1"], @(-1));
+            [x fulfill];
+        }
+    }];
+    
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        [self createDocNumbered: -1 of: 100];
+    });
+    
+    [self waitForExpectationsWithTimeout: 10.0 handler: ^(NSError *error) { }];
+    
+    [q removeChangeListenerWithToken: token];
+}
+
+- (void) testLiveQueryNoUpdate {
+    [self loadNumbers: 100];
+    
+    __block int count = 0;
+    CBLQuery* q = [CBLQueryBuilder select: @[]
+                                     from: [CBLQueryDataSource database: self.db]
+                                    where: [[CBLQueryExpression property: @"number1"] lessThan: [CBLQueryExpression integer: 10]]
+                                  orderBy: @[[CBLQueryOrdering property: @"number1"]]];
+    
+    id token = [q addChangeListener:^(CBLQueryChange* change) {
+        count++;
+        AssertNotNil(change.query);
+        AssertNil(change.error);
+        NSArray<CBLQueryResult*>* rows = [change.results allObjects];
+        if (count == 1) {
+            AssertEqual(rows.count, 9u);
+        } else {
+            XCTFail(@"Unexpected update from LiveQuery");
+        }
+    }];
+    
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)),
```

Comment:
Same comment as above.

Reply:
This cannot be done as previous. we are waiting for next 2 secs to make sure, there are no changes notified with this change. So it's better to keep it as before.